### DISCUSSION TITLE
Create a virtual math keyboard with KaTeX + MathKeyboardEngine

### DISCUSSION THREAD

**Discussion by SymboLinker** [KaTeX/KaTeX#3530] [Show and tell]
Hi all,

I've published a JavaScript library that you can use to create a virtual math keyboard. It is intended for use together with any LaTeX typesetting library (for example MathJax or KaTeX).

GitHub repository: https://github.com/MathKeyboardEngine/MathKeyboardEngine
Live examples: https://mathkeyboardengine.github.io

My goal was to create an easy-to-use virtual math keyboard library that is highly customizable. I hope you find it interesting and useful!

Symbolinker

**Comment by edemaine** [comment]
Very cool! Love the LaTeX output.

Personally I'd like to be able to type LaTeX commands I know, like `\times` (maybe followed by a space), and have it get translated into the equivalent of hitting the corresponding button; currently the `\` seems to be ignored in the LaTeX. This is how it works in Microsoft Office's math mode, for example.

**Reply by SymboLinker** [reply]
Hi Erik,

I really appreciate your feedback and because of it I've just added (a minimal form of) the functionality to the "Live examples". Typing LaTeX commands is supported in the examples if they are written with [a-zA-Z] and if they do not have [placeholders](https://mathkeyboardengine.github.io/docs/0.1/#Placeholder). The library can handle more, but programmers can now get a glimpse of what is possible and that may be enough for creating the functionality they like.

Thanks again,
Symbolinker
